shotover/src/transforms/tee.rs

```diff
@@ -198,6 +203,24 @@ impl Transform for Tee {
 
                 Ok(chain_response)
             }
+            ConsistencyBehavior::LogMismatch => {
+                let (tee_result, chain_result) = tokio::join!(
+                    self.tx
+                        .process_request(requests_wrapper.clone(), self.timeout_micros),
+                    requests_wrapper.call_next_transform()
+                );
+
+                let tee_response = tee_result?;
+                let chain_response = chain_result?;
+
+                if !chain_response.eq(&tee_response) {
+                    warn!(
+                        "Tee mismatch: chain response: {:?} tee response: {:?}",
```

Comment:
A new line should help readability:
`Tee mismatch: chain response: {:?}\ntee response: {:?}`

Reply:
We should probably also generate a better message, the message inner stuff should not be exposed to the user.
But if we just call to_high_level_string that would be expensive for messages that arent already parsed.
Maybe we should just log the raw message bytes or maybe we should avoid including the message at all?

For now I would say use to_high_level_string, if this is a performance issue for anyone they can always use `ConsistencyBehaviour::Ignore`

Reply:
>to_high_level_string that would be expensive for messages

This will only occur on mismatches so I think the performance hit is fine, assuming a mismatch is an error and not part of some expected behaviour when using the Tee transform

Reply:
If errors are significantly more expensive to handle than regular messages, if say 50% of messages start to error that could be enough to cause shotover to fail to serve the remaining 50% of successful messages.
Thats what I want to be careful of here.

That said since this is one of multiple configuration modes here its probably fine to take the performance hit for more meaningful warnings.
If the user wants no performance hit on mismatch they can use `Ignore`
I'm happy to go ahead with this as is. (just need to fix the small other comment I made)